libftl/handshake.c

```diff
@@ -114,10 +114,43 @@ ftl_status_t _init_control_connection(ftl_stream_configuration_private_t *ftl) {
   }
 
   ftl->ingest_socket = sock;
-  
+
   return FTL_SUCCESS;
 }
 
+ftl_status_t _test_stream_key_internal(ftl_stream_configuration_private_t *ftl)
```

Comment:
Is there a reason you can't just do this while you connect? I only ask because we want to try to minimize the number of connections to ingest, so I don't know if we want to be making a connection just to check the key.